common/src/main/scala/org/apache/spark/sql/comet/util/Utils.scala

```diff
@@ -261,4 +262,32 @@ object Utils {
         throw new SparkException(s"Unsupported Arrow Vector for $reason: ${valueVector.getClass}")
     }
   }
+
+  // Given an array of Comet vectors calculate the number of bytes required for a batch of
+  // UnsafeRow rows to hold the values of the vectors
+  def getUnsafeRowBatchSize(vectors: Array[CometVector]): Long = {
+    val bitSetWidth = UnsafeRow.calculateBitSetWidthInBytes(vectors.length)
+    val num_rows = vectors.apply(0).getValueVector.getValueCount
+    val dataBytes: Long = vectors
+      .map(v => {
+        val dt = fromArrowField(v.getValueVector.getField)
+        assert(
+          UnsafeRow.isMutable(dt) || dt.isInstanceOf[BinaryType] || dt.isInstanceOf[StringType])
+        // For variable length types, assuming that the vector has not been read from, the
+        // readable bytes are the number of bytes of data in the vector.
+        val fixedBytes = num_rows * 8L // offset (4 bytes) and length (4 bytes)
+        val varBytes = dt match {
+          case datatype if UnsafeRow.isFixedLength(datatype) => 0L
+          case DecimalType.Fixed(_, _) => num_rows * 16L
+          case BinaryType | StringType =>
```

Comment:
This case is currently unused since only the fixed length types are implemented. This will probably change completely.